### PR TITLE
`General`: Improve layout of exercise management overview

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

Improve the layout of the exercise manage over view to use the space more efficiently.

### Description

The course header is now even shorter still including longer descriptions by reducing font size of the description.
Move headlines and create exercise buttons in the exercise overview (see screenshot)

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student

1. Log in to Artemis
2. Navigate to Course Administration
3. Check that the course header looks good with different screen sizes, descriptions lengths and with and without a course icon.
4. Check that the exercise management overview for the instructor looks good and displays all relevant information
5. Check that creating a programming, quiz and modeling/text/fileupload (one of the three) still works

#### Performance Review
- [x] I confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

image

### CODE REVIEW

**Comment by JohannesStoehr** [comment]
> Tested on ts1. Everything looks good with/without picture and with long/short description. Creating and importing exercises also works.

Thanks for testing!

> I had one issue: When completely deleting the course description, the header does not update until you exit course administration and enter the course again. On the other hand, when changing the course description, the update is shown immediately.

This is unrelated to this PR and is also present on develop. So it should not hinder merging this PR

**Review by SolizerCodes — DISMISSED** [review]

**Review by eceeeren — APPROVED** [review]
Tested on ts2. The issue that @SolizerCodes mentioned will be ignored. Everything else works as expected.

**Review by fschoenberger — CHANGES_REQUESTED** [review]
Deployed to TS1. When logging in as instructor or student I get many internal server errors.

File: src/main/webapp/app/exercises/programming/manage/programming-exercise.component.html [hunk 1]
```diff
@@ -1,297 +1,248 @@
...
-                                <a href="{{ programmingExercise.templateParticipation!.repositoryUrl! }}" target="_blank">Template</a></span
+                        <jhi-programming-exercise-instructor-status
+                            *ngIf="programmingExercise.solutionParticipation?.results?.length"
+                            [participationType]="solutionParticipationType"
+                            [participation]="programmingExercise.solutionParticipation!"
+                            [exercise]="programmingExercise"
+                        ></jhi-programming-exercise-instructor-status>
+                    </div>
+                    <div>
+                        <span *ngIf="programmingExercise.testRepositoryUrl"><a href="{{ programmingExercise.testRepositoryUrl }}" target="_blank">Test</a></span>
```
[the remark this answers is not in the document]
Comment (JohannesStoehr) [reply src/main/webapp/app/exercises/programming/manage/programming-exercise.component.html:83]: This only links to our own repositories and is fine

**Review by RY997 — DISMISSED** [review]
Tested on ts2, worked as expected

**Review by SabrinaGlatz — DISMISSED** [review]
Tested on TS1. Works as expected.

**Review by SolizerCodes — DISMISSED** [review]

**Review by dakennguyen — CHANGES_REQUESTED** [review]
tested on ts2, the layout is good, but the server responds with 500 when I try to update a long title. Could you pls confirm if it's relevant?

**Review by greekgeek1 — APPROVED** [review]
Tested on ts3. Works as expected. Could not recreate the error observed by dakennguyen.

**Review by dakennguyen — DISMISSED** [review]
> tested on ts2, the layout is good, but the server responds with 500 when I try to update a long title. Could you pls confirm if it's relevant?

It works now on ts1. I think it was because of the internal server error that we have on develop branch

**Review by MaximilianAnzinger — DISMISSED** [review]
Code looks good to me.

The `Potentially unsafe external link` Warnings seem to be insignificant since the links are internal.

**Comment by bassner** [comment]
@JohannesStoehr We have some conflicts in this PR

**Review by MaximilianAnzinger — APPROVED** [review]
Code changes look good to me

**Review by tobias-lippert — APPROVED** [review]
Tested on ts2. 
Course header and course management overview look both good (with and without a course icon)
Creating a text exercise is still possible.

**Review by eceeeren — APPROVED** [review]
Tested on TS2, works as expected.
However it still gives an Internal Server Error when adding a very long title, this is not related to this PR right?

**Review by asliayk — APPROVED** [review]
tested on ts1, course header and exercise management layouts look as expected with an icon or without an icon. I could also create a programming exercise.

**Pull request merged**